File: application/lib/models/family_member.dart
class FamilyMember {
  final int id;
  final String name;
  final String photoUrl;
  final bool faceEnrolled;
  final DateTime createdAt;

  FamilyMember({
    required this.id,
    required this.name,
    required this.photoUrl,
    required this.faceEnrolled,
    required this.createdAt,
  });

  factory FamilyMember.fromJson(Map<String, dynamic> json) {
    return FamilyMember(
      id: json['id'] as int,
      name: json['name'] as String,
      photoUrl: (json['photo_url'] as String?) ?? '',
      faceEnrolled: (json['face_enrolled'] as bool?) ?? false,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}

File: backend/internal/models/family_member.go
package models

import "time"

type FamilyMember struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	UserID       uint      `gorm:"not null" json:"user_id"`
	Name         string    `gorm:"uniqueIndex:idx_family_member_user_name;not null" json:"name"`
	PhotoURL     string    `json:"photo_url"`
	FaceEnrolled bool      `gorm:"default:false" json:"face_enrolled"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}
